kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -1301,26 +1301,26 @@ unifyEqualsElementLists
                         "Duplicated element in unification results"
                         first
                         second
-                Just remainderTerm -> case opaque of
-                    ElemVar_ _
-                        | TermLike.isFunctionPattern remainderTerm -> do
-                            opaqueUnifier <- unifyEqualsChildren opaque remainderTerm
-                            let (opaqueTerm, opaqueCondition) =
-                                    Pattern.splitTerm opaqueUnifier
-                                result = unifier `andCondition` opaqueCondition
-
-                            return (result, [opaqueTerm])
-                    _ ->
-                        error . show . Pretty.vsep $
-                            [ "Unification case that should be handled somewhere else: \
-                              \attempting normalized unification with a \
-                              \non-element-variable opaque term or \
-                              \non-function maps could lead to infinite loops."
-                            , Pretty.indent 2 "first="
-                            , Pretty.indent 4 (unparse first)
-                            , Pretty.indent 2 "second="
-                            , Pretty.indent 4 (unparse second)
-                            ]
+                Just remainderTerm
+                    | TermLike.isFunctionPattern remainderTerm -> do
+                        opaqueUnifier <-
+                            unifyEqualsChildren
+                                (mkElemVar opaqueElemVar)
+                                remainderTerm
+                        let (opaqueTerm, opaqueCondition) =
+                                Pattern.splitTerm opaqueUnifier
+                            result = unifier `andCondition` opaqueCondition
+                        return (result, [opaqueTerm])
+                _ ->
+                    error . show . Pretty.vsep $
+                        [ "Unification case that should be handled somewhere else: \
+                          \attempting normalized unification with \
+                          \non-function maps could lead to infinite loops."
+                        , Pretty.indent 2 "first="
+                        , Pretty.indent 4 (unparse first)
+                        , Pretty.indent 2 "second="
+                        , Pretty.indent 4 (unparse second)
+                        ]
```

Comment:
Why is this still reachable?

Reply:
`remainderTerm` may not be a function pattern and as the error message suggests, attempting normalized unification with non-function maps could lead to infinite loops.